doc/build/orm/extensions/associationproxy.rst

```diff
@@ -229,18 +233,23 @@ Where above, each ``.keywords.append()`` operation is equivalent to::
 
     >>> user.user_keywords.append(UserKeyword(Keyword('its_heavy')))
 
-The ``UserKeyword`` association object has two attributes here which are populated;
-the ``.keyword`` attribute is populated directly as a result of passing
-the ``Keyword`` object as the first argument.   The ``.user`` argument is then
-assigned as the ``UserKeyword`` object is appended to the ``User.user_keywords``
-collection, where the bidirectional relationship configured between ``User.user_keywords``
-and ``UserKeyword.user`` results in a population of the ``UserKeyword.user`` attribute.
-The ``special_key`` argument above is left at its default value of ``None``.
+The ``UserKeyword`` association object has two attributes which are
+populated, with each attribute populated at a different stage of the
+operation; the ``.keyword`` attribute is populated directly as a
+result of passing the ``Keyword`` object as the first argument and
+this happens via the ``user.keywords.append()`` operation.  The
+``.user`` argument is then assigned during the step where the
+``UserKeyword`` object is appended to the ``User.user_keywords``
+collection, wherein the bidirectional relationship configured between
+``User.user_keywords`` and ``UserKeyword.user`` results in a
+population of the ``UserKeyword.user`` attribute.  The ``special_key``
+argument above is left at its default value of ``None``.
 
 For those cases where we do want ``special_key`` to have a value, we
-create the ``UserKeyword`` object explicitly.  Below we assign all three
-attributes, where the assignment of ``.user`` has the effect of the ``UserKeyword``
-being appended to the ``User.user_keywords`` collection::
+create the ``UserKeyword`` object explicitly.  Below we assign all
```

Comment:
"In the example below, we illustrate all three attributes being assigned explicitly.  The assignment of ``UserKeyword.user``, thanks to the back_populates handler referring to ``User.user_keywords``, has the effect of the ``UserKeyword`` object being appended to the ``User.user_keywords`` collection."

this example also gets into a behavior that has changed in 1.4/2.0 which is that UserKeyword won't be cascaded into the User object's Session if User is already attached.  However there does not seem to be a section in the 2.0 documentation regarding this;  the previous section is https://docs.sqlalchemy.org/en/14/orm/backref.html#setting-cascade-for-backrefs and is 1.x specific.  While I removed this section for 2.0, I should restore it and just adapt it for 2.0's behavior as there's otherwise no mention of it.  Then this section can also refer to it as a seealso.